### PR TITLE
LG-3984 Expose virtualizer options in `useLeafyGreenTable` options

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Exposes `virtualizerOptions` property in `useLeafyGreenTable` options. This is used to define options for `react-virtual`.

As of now, we have `react-virtual@2.10.4` as out virtualizer dependency. Virtualizer options for this version can be found [on GitHub](https://github.com/TanStack/virtual/blob/v2/docs/src/pages/docs/api.md) 

This resolves the linked bug by allowing consumers to provide a `estimateSize` callback which can be used to estimate the size of a virtual row when they do not match the default row height (`40px`)

🎟 _Jira ticket:_ [LG-3984](https://jira.mongodb.org/browse/LG-3984)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/table/src/Table/TableWithVS.story.tsx [hunk 1]
```diff
@@ -49,52 +50,54 @@ const meta: StoryMetaType<typeof Table> = {
 export default meta;
 
 const virtualScrollingContainerHeight = css`
-  height: 500px;
+  /* height: 500px; */
```
Comment (bruugey) [inline packages/table/src/Table/TableWithVS.story.tsx:53]: these both feel like magic numbers?

Comment (bruugey) [reply packages/table/src/Table/TableWithVS.story.tsx:53]: i guess its in the story file

Comment (TheSonOfThomp) [reply packages/table/src/Table/TableWithVS.story.tsx:53]: Yeah, they're arbitrary but it's the story file so it's not a huge deal

**Comment by khelif96** [comment]
This change makes sense to me. I left a comment on the ticket. I'm going to remove myself as a reviewer.

File: .changeset/thick-dragons-vanish.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@leafygreen-ui/table': minor
+---
+
+Exposes `virtualizerOptions` property in `useLeafyGreenTable` options. This is used to define options for `react-virtual`.
```
Comment (shaneeza) [inline .changeset/thick-dragons-vanish.md:5]: Do you also want to link the ticket as a reference?

File: packages/table/src/Table/TableWithVS.story.tsx [hunk 3]
```diff
@@ -118,7 +120,7 @@ export const Basic: StoryFn<StoryTableProps> = args => {
         ref={tableContainerRef}
         className={virtualScrollingContainerHeight}
       >
-        <TableHead>
+        <TableHead isSticky>
```
Comment (shaneeza) [inline packages/table/src/Table/TableWithVS.story.tsx:123]: why the change to `isSticky`?

File: packages/table/src/useLeafyGreenTable/ReactVirtual.types.ts [hunk 4]
```diff
@@ -1,20 +1,5 @@
-import { VirtualItem } from 'react-virtual';
+import { useVirtual } from 'react-virtual';
```
Comment (shaneeza) [inline packages/table/src/useLeafyGreenTable/ReactVirtual.types.ts:1]: Whats the diff between `VirtualItem` and `useVirtual`?

File: packages/table/src/Table/TableWithVS.story.tsx (hunk 3 above)
Comment (TheSonOfThomp) [reply packages/table/src/Table/TableWithVS.story.tsx:123]: Doesn't necessarily need to be on all stories, but IMO it's a better demo experience when the headers don't scroll out of view

Comment (shaneeza) [reply packages/table/src/Table/TableWithVS.story.tsx:123]: Screenshot 2024-02-21 at 10 36 24 AM

 I think this is causing z-index issues. The nested arrow is above the header. The same happens with checkboxes.

File: packages/table/src/Table/TableWithVS.story.tsx [hunk 5]
```diff
@@ -651,3 +539,91 @@ export const ExpandableContent: StoryFn<StoryTableProps> = args => {
     </>
   );
 };
+
+export const TallRows: StoryFn<StoryTableProps> = args => {
```
Comment (shaneeza) [inline packages/table/src/Table/TableWithVS.story.tsx:543]: In this story, you can continue to scroll even when there is no more rows. Is this expected? 

https://jmp.sh/mdpgthXk

In this video, I'm scrolling down and it looks like the scrollbar continues to move when there are no more rows.

File: packages/table/src/useLeafyGreenTable/useLeafyGreenTable.types.ts [hunk 6]
```diff
@@ -38,23 +41,31 @@ export type LGColumnDef<
...
-  containerRef: RefObject<HTMLDivElement>;
+> = Omit<TableOptions<LGTableDataType<T>>, 'getCoreRowModel' | 'columns'> & {
+  containerRef: RefObject<HTMLElement>;
   hasSelectableRows?: boolean;
-  useVirtualScrolling?: boolean;
   columns: Array<LGColumnDef<T, V>>;
   withPagination?: boolean;
   allowSelectAll?: boolean;
-}
+} & {
```
[the remark this answers is not in the document]
Comment (TheSonOfThomp) [reply packages/table/src/useLeafyGreenTable/useLeafyGreenTable.types.ts:58]: Ah, it's not anymore 👍

File: packages/table/src/Table/TableWithVS.story.tsx (hunk 3 above)
Comment (TheSonOfThomp) [reply packages/table/src/Table/TableWithVS.story.tsx:123]: Hmm. Opened this ticket https://jira.mongodb.org/browse/LG-4037

Comment (TheSonOfThomp) [reply packages/table/src/Table/TableWithVS.story.tsx:123]: Resolved 👍

**Review by shaneeza — APPROVED** [review]
